tests/rollout/test_model_max_lens.py

```
@@ -0,0 +1,137 @@
+        )
+
+        # Offload vllm model to reduce peak memory usage
+        self.inference_engine.sleep(level=1)
+
+        kwargs = dict(
+            n=1,
+            logprobs=0,  # can be set to 0 and let actor to recompute
+            max_tokens=config.response_length,
+        )
+
+        # # we may detokenize the result all together later
+        if vllm_version != '0.3.1':
+            kwargs['detokenize'] = False
+
+        # supporting adding any sampling params from the config file
+        for k in config.keys():
+            if hasattr(SamplingParams(), str(k)):
+                kwargs[k] = config.get(k)
+
+        print(f"kwargs: {kwargs}")
+        self.sampling_params = SamplingParams(**kwargs)
+
+        self.pad_token_id = tokenizer.pad_token_id
+
+    @torch.no_grad()
+    def generate_sequences(self):
+        # this function is not used in the initialization of the rollout object
+        # thus passed in this test
+        pass
+
+def test_vllm_rollout_with_yarn_position_embeddings():
+    """
+    Test the vLLM rollout with yarn position embeddings.
+    """
+
+    config = OmegaConf.create({
+        'model_path': "OldKingMeister/Qwen2.5-1.5B-Instruct-YaRN",
+        'prompt_length': 32768,
+        'response_length': 512,
+        'dtype': 'bfloat16',
+        'enforce_eager': False,
+        'gpu_memory_utilization': 0.9,
+        'enable_chunked_prefill': True,
+        'free_cache_engine': False,
+        'disable_log_stats': True, 
+    })
+
+    tokenizer = AutoTokenizer.from_pretrained(config.model_path)
+    model_hf_config = AutoConfig.from_pretrained(config.model_path)
+
+    rollout = vLLMRollout(
+        model_path=config.model_path,
+        config=config,
+        tokenizer=tokenizer, 
+        model_hf_config=model_hf_config,
+    )
+
+if __name__ == "__main__":
+    import os 
+    os.environ['VLLM_WORKER_MULTIPROC_METHOD'] = 'spawn'
+    try:
+        test_vllm_rollout_with_yarn_position_embeddings()
```

Could u remove the try catch? We want to let it fail if the functionality breaks in the future

---

Sure, I'll remove the try catch asap.